File: automator/src/main/java/com/stardust/automator/filter/StringEndsWithFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class StringEndsWithFilter extends DfsFilter {

    private final String mSuffix;
    private final KeyGetter mKeyGetter;

    public StringEndsWithFilter(String suffix, KeyGetter keyGetter) {
        mSuffix = suffix;
        mKeyGetter = keyGetter;
    }

    @Override
    protected boolean isIncluded(UiObject nodeInfo) {
        String key = mKeyGetter.getKey(nodeInfo);
        return key != null && key.endsWith(mSuffix);
    }

    @Override
    public String toString() {
        return mKeyGetter.toString() + "EndsWith(\"" + mSuffix + "\")";
    }
}

File: automator/src/main/java/com/stardust/automator/filter/StringEndsWithFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

class StringEndsWithFilter(private val mSuffix: String, private val mKeyGetter: KeyGetter) : Filter {

    override fun filter(node: UiObject): Boolean {
        val key = mKeyGetter.getKey(node)
        return key != null && key.endsWith(mSuffix)
    }

    override fun toString(): String {
        return mKeyGetter.toString() + "EndsWith(\"" + mSuffix + "\")"
    }
}
